submissions/LuckyDnepr/memory-pair-game/script.js

```diff
@@ -0,0 +1,139 @@
+const doc = document;
+
+const themesDataPath = "./themesData.json";
+
+let themesData,
+    themeIndex = 0,
+    numberOfCards = 6,
+    cardsSet,
+    openedCards = [],
+    foundPairs;
+
+doc.querySelector(".game_field").addEventListener("click", (e) => {
+    if (e.target.classList.contains("card_face")) {
+        cardFlip(e.target.parentNode);
+    }
+});
+
+makeMenuAndListeners(themesDataPath);
+
+function gameStart() {
+    foundPairs = 0;
+    renderGameField(numberOfCards);
+    generateCardsSet(numberOfCards, themesData[themeIndex].pictures.length - 1);
+    switchMenuAndGameField();
+}
+
+function renderGameField(numberOfCards) {
+    const gameField = doc.querySelector(".game_field");
+    switch (numberOfCards) {
+        case 6:
+            gameField.setAttribute("style", "--cols: 3; --rows: 2");
+            break;
+        case 12:
+            gameField.setAttribute("style", "--cols: 4; --rows: 3");
+            break;
+        default:
+            break;
+    }
+    gameField.innerHTML = new Array(numberOfCards)
+        .fill(0)
+        .reduce((innerhtml, curr,  i) => {
+            return innerhtml +=`<div class="game_card" data-card_id="${i}">
+                <img src="./images/bg-hearts.png" alt="" class="card_face card_face_front">
+                <img src="#" alt="" class="card_face card_face_back">
+            </div>
+        `;
+        }, "");
+}
+
+function cardFlip(card) {
+    if (!card.classList.contains("flipped")) {
+        card.classList.add("flipped");
+        openedCards.push(card);
+        changeCardFace(card);
+        checkPair();
+    }
+}
+
+function checkPair() {
+    if (openedCards.length % 2 == 0) {
```

Comment:
```suggestion
    if (openedCards.length === 2) {
```
Your array.length never will be greater than 2